Odysee/Utils/Extensions.swift

```diff
@@ -62,3 +62,28 @@ extension UIApplication {
         return viewController
     }
 }
+
+extension URLSession {
+    struct DataTaskSuccess {
+        var data: Data
+        var response: URLResponse
+    }
+
+    // A convenience wrapper for dataTask() that gives a Result instead of three optionals.
+    func dataTask(with request: URLRequest,
+                  completionHandler: @escaping (Result<DataTaskSuccess, Error>) -> Void) -> URLSessionDataTask {
+        return self.dataTask(with: request) { data, response, error in
+            let result = Result<DataTaskSuccess, Error> {
+                if let error = error {
+                    throw error
+                }
+                guard let data = data, let response = response else {
+                    assertionFailure()
+                    throw GenericError("no error but no data and/or no response: '\(data.debugDescription)' '\(response.debugDescription)")
```

Comment:
Tired to build, but I get an error at this point: `.../Extensions.swift:82:112: Variable declared in 'guard' condition is not usable in its body`

Reply:
Looks like another compiler version issue. I think the error is incorrect – I'm trying to using the _optional_ version of those two variables inside the body, which should be allowed. If it's just you and me working on it, WDYT about moving to Xcode 12.5?

Reply:
I'll remove them from the log line to get us unblocked.